src/DependencyManagement/DependencyManager.cs

```diff
@@ -185,58 +188,86 @@ internal void Initialize(FunctionLoadRequest request)
+                    string latestVersion = module.LatestVersion;
+
+                    int tries = 1;
+
+                    while (true)
                     {
-                        string moduleName = module.Name;
-                        string latestVersion = module.LatestVersion;
-
-                        // Save the module to the given path
-                        pwsh.AddCommand("PowerShellGet\\Save-Module")
-                            .AddParameter("Repository", Repository)
-                            .AddParameter("Name", moduleName)
-                            .AddParameter("RequiredVersion", latestVersion)
-                            .AddParameter("Path", DependenciesPath)
-                            .AddParameter("Force", Utils.BoxedTrue)
-                            .AddParameter("ErrorAction", "Stop")
-                            .InvokeAndClearCommands();
-
-                        var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, latestVersion);
-                        logger.Log(LogLevel.Trace, message, isUserLog: true);
+                        try
+                        {
+                            // Save the module to the given path
+                            pwsh.AddCommand("PowerShellGet\\Save-Module")
+                                .AddParameter("Repository", Repository)
+                                .AddParameter("Name", moduleName)
+                                .AddParameter("RequiredVersion", latestVersion)
+                                .AddParameter("Path", DependenciesPath)
+                                .AddParameter("Force", Utils.BoxedTrue)
+                                .AddParameter("ErrorAction", "Stop")
+                                .InvokeAndClearCommands();
+
+                            var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, latestVersion);
+                            logger.Log(LogLevel.Trace, message, isUserLog: true);
+
+                            break;
+                        }
+                        catch (Exception e)
+                        {
+                            if (tries >= MaxNumberOfTries)
+                            {
+                                var errorMsg = string.Format(PowerShellWorkerStrings.FailToInstallFuncAppDependencies, e.Message);
+                                _dependencyError = new DependencyInstallationException(errorMsg, e);
+
+                                throw _dependencyError;
+                            }
+
+                            // Update the retry counter
+                            tries++;
+                        }
+
+                        // Wait for 2^(tries-1) seconds between retries. In this case, it would be 1, 2, and 4 seconds, respectively.
+                        double waitTimeInSeconds = Math.Pow(2, tries - 1);
+                        TimeSpan.FromSeconds(waitTimeInSeconds);
```

Comment:
There is no wait happening here: this line just calculates the timespan and discards it.

Reply:
Thanks for pointing this out. I lost a line of code when doing the three way merge. Fixed.